libirods_rule_engine_plugin-audit_amqp.cpp

```diff
@@ -127,35 +129,30 @@ irods::error start(irods::default_re_ctx& _u,const std::string& _instance_name)
     pn_messenger_start(messenger);
     pn_messenger_set_blocking(messenger, false);  // do not block
     
-    json_t* obj = json_object();
-    if( !obj ) {
-        return ERROR(SYS_MALLOC_ERR, "json_object() failed");
-    }
+    nlohmann::json json_obj;
 
     struct timeval tv;
     gettimeofday(&tv, NULL);
     unsigned long time_ms = tv.tv_sec * 1000 + tv.tv_usec / 1000;
     std::stringstream time_str; time_str << time_ms;
-    json_object_set(obj, "time_stamp", json_string(time_str.str().c_str()));
+    json_obj["time_stamp"] = time_str.str(); 
 
     char host_name[MAX_NAME_LEN];
     gethostname( host_name, MAX_NAME_LEN );
-    json_object_set(obj, "hostname", json_string(host_name));
+    json_obj["hostname"] = std::string(host_name);
 
     pid_t pid = getpid();
     std::stringstream pid_str; pid_str << pid;
-    json_object_set(obj, "pid", json_string(pid_str.str().c_str()));
+    json_obj["pid"] = pid_str.str();
 
-    json_object_set(obj, "action", json_string("START"));
+    json_obj["action"] = "START";
 
     std::string log_file;
     if (test_mode) {
-        log_file = str(boost::format("%s/%06i.txt") % log_path_prefix % pid);
-        json_object_set(obj, "log_file", json_string(log_file.c_str()));
+	json_obj["log_file"] = str(boost::format("%s/%06i.txt") % log_path_prefix % pid);
     }
 
-    char* tmp_buf = json_dumps( obj, JSON_INDENT( 0 ) );
-    std::string msg_str = std::string("__BEGIN_JSON__") + std::string(tmp_buf) + std::string("__END_JSON__");
+    std::string msg_str str(boost::format() %  );
```

Comment:
Something bad happened here... probably want this to be like this instead?
```c++
    std::string msg_str = std::string("__BEGIN_JSON__") + json_obj.dump() + std::string("__END_JSON__");
```

Reply:
You're very right about the copy-paste antics; this in particular was the fruit of some minor stage fright.